parsers/src/main/java/org/iqu/parsers/HtmlEventsParser.java

```diff
@@ -35,77 +36,83 @@
-	private static DateFormat dF = new SimpleDateFormat(pattern);
-	private String categories = "";
-	private String source;
-
-	@Override
-	public List<Event> readFeed(String sourceURL, String encoding) {
-
-		List<Event> events = new ArrayList<Event>();
-		source = sourceURL;
-		Document doc = null;
-		try {
-			doc = Jsoup.connect(sourceURL).get();
-		} catch (IOException e) {
-			LOGGER.error("Error loading URL");
-		}
-		Elements item = doc.getElementsByClass(EVENTGROUP);
-
-		for (Element element : item) {
-
-			String UrlToConnect = element.getElementsByTag(A).attr(URL);
-			if (UrlToConnect != null && UrlToConnect.length() != 0)
-				takeEvents(element.getElementsByTag(A).attr(URL), events);
-		}
-
-		return events;
-	}
-
-	private void takeEvents(String eventsURL, List<Event> events) {
-		Document doc = null;
-		try {
-			doc = Jsoup.connect(eventsURL).get();
-		} catch (MalformedURLException e) {
-			LOGGER.error("Invalid URL!", e);
-		} catch (IOException e) {
-			LOGGER.error("InputStream error!", e);
-		}
-		categories = doc.body().id();
-		Elements item = doc.getElementsByClass(EVENTLIST);
-		for (Element element : item) {
-			event = new Event();
-			try {
-				getTags(element);
-				events.add(event);
-			} catch (ParseException e) {
-				LOGGER.error("Invalid tags");
-			}
-		}
-
-	}
-
-	private void getTags(Element element) throws ParseException {
-		event.setTitle(element.getElementsByTag(A).attr(TITLE));
-		event.setImage_id(element.getElementsByTag(IMG).attr(IMGSRC));
-		event.setExternal_url(element.getElementsByTag("a").attr(URL));
-		event.setDescription(element.getElementsByClass(DESCRIPTION).html());
-		event.setId(element.getElementsByClass(EVENTLIST).attr(ID) + "-" + categories);
-		event.setStartDate(convertDate(element.getElementsByTag(TIME).attr(DATE)));
-		event.setSource(source);
-		event.setCategories(categories);
-		event.setEndDate(
-				convertDate(element.getElementsByAttributeValue("itemprop", "endDate").attr(HtmlParserConstants.DATE)));
-	}
-
-	private long convertDate(String date) throws ParseException {
-		if (date.length() > 0) {
-			return dF.parse(date).getTime();
-		}
-		return 0;
-	}
+  private Event event;
+  private static final Logger LOGGER = Logger.getLogger(HtmlEventsParser.class);
+  private static String pattern = "yyyy-MM-dd'T'HH:mm:ss'+'hh:mm";
+  private static DateFormat dF = new SimpleDateFormat(pattern);
+  private String categories = "";
+  private String source;
+
+  @Override
+  public Source getSource() {
+    // TODO: implement reading of source information from feed
+    return null;
+  }
+
+  @Override
+  public List<Event> readFeed(String sourceURL, String encoding) {
+
+    List<Event> events = new ArrayList<Event>();
+    source = sourceURL;
+    Document doc = null;
+    try {
+      doc = Jsoup.connect(sourceURL).get();
+    } catch (IOException e) {
+      LOGGER.error("Error loading URL");
+    }
+    Elements item = doc.getElementsByClass(EVENTGROUP);
```

Comment:
This line might throw a NullPointerException if there's an exception thrown at line 59.